### ISSUE TITLE
pPrint appears to go into an infinite loop

### ISSUE THREAD

**Issue by cdepillabout** [cdepillabout/pretty-simple#28]
Trying to pretty print the following causes `pPrint` to appear to go into an infinite loop:

```haskell
Effect (Identity (Step (1 :> Effect (Identity (Step (2 :> Effect (Identity (Step (1 :> Effect (Identity (Step (3 :> Effect (Identity (Step (2 :> Effect (Identity (Step (1 :> Effect (Identity (Step (4 :> Effect (Identity (Step (3 :> Effect (Identity (Step (2 :> Effect (Identity (Step (1 :> Effect (Identity (Return ()))))))))))))))))))))))))))))))))
```

**Comment by cdepillabout** [comment]
Actually, this could just be the function taking too long to run, as shown in #30?

**Comment by cdepillabout** [comment]
Fixed by #32.
